### PR TITLE
fix rock5itx rtc detect failure

### PR DESCRIPTION
**Pull request by Ken-Vamrs**
rtc: hym8563: Solve the problem of probabilistic initialization failure of hym8563

### CODE REVIEW

File: drivers/rtc/rtc-hym8563.c [hunk 1]
```diff
@@ -473,9 +474,15 @@ static irqreturn_t hym8563_irq(int irq, void *dev_id)
...
 	/* Clear stop flag if present */
-	ret = i2c_smbus_write_byte_data(client, HYM8563_CTL1, 0);
+	for(;retry > 0;retry--){
+		ret = i2c_smbus_write_byte_data(client, HYM8563_CTL1, 0);
+		if(ret >= 0)
+			break;
+		msleep(10);
+	}
 	if (ret < 0)
 		return ret;
```
Comment (RadxaYuntian) [inline drivers/rtc/rtc-hym8563.c]: 这里return -EPROBE_DEFER来deferred probe不行么？这个明显是初始化顺序的问题

Comment (Ken-Vamrs) [reply drivers/rtc/rtc-hym8563.c]: 验证ok，第一次init失败后defer probe重新probe成功
[    8.861960] =========》ken  W rtc
[    8.861965] rtc-hym8563 6-0051: could not init device, -517
[    9.652504] rtc-hym8563 6-0051: rtc information is valid
[    9.658953] rtc-hym8563 6-0051: registered as rtc0
[    9.660171] rtc-hym8563 6-0051: setting system clock to 2024-04-10T09:25:56 UTC (1712741156)

**Review by RadxaYuntian — APPROVED** [review]
我可能会选择在hym8563_probe这里调用hym8563_init_device时返回-EPROBE_DEFER，但如果ken确定这样就可以的话也行

**Comment by Ken-Vamrs** [comment]
> 我可能会选择在hym8563_probe这里调用hym8563_init_device时返回-EPROBE_DEFER，但如果ken确定这样就可以的话也行

确实这样是改动最少的，我等会再验证一下

**Comment by Ken-Vamrs** [comment]
> > 我可能会选择在hym8563_probe这里调用hym8563_init_device时返回-EPROBE_DEFER，但如果ken确定这样就可以的话也行
> 
> 确实这样是改动最少的，我等会再验证一下

验证通过，这个初始化失败一般是在冷启动时出现概率比较大
[    8.747272] rtc-hym8563 6-0051: kenken ===> could not init device, -6
[    9.557798] rtc-hym8563 6-0051: rtc information is valid
[    9.564234] rtc-hym8563 6-0051: registered as rtc0
[    9.565697] rtc-hym8563 6-0051: setting system clock to 2024-04-11T06:31:25 UTC (1712817085)

**Pull request merged**